### ISSUE TITLE
Have single script editing and running environment.

### ISSUE THREAD

**Issue by abalter** [imagej/imagej-legacy#133]
Currently, scripts run differently in Plugins>Macros>run and from the script editor (currently only accessible through File>New>Script). The script editor is more consistent because for Javascript macros it avoids the automatic imports done in the Javascript-like macro language. That causes some confusion.

The proposal is that the features of the two script editors and running environments be combined. The script editor is a much more sophisticated and user-friendly editor. The only thing it lacks as far as I can see is debugging. It also runs Javascript macros in a clean environment.

**Comment by ctrueden** [comment]
Actually, ImageJ2's Script Editor pops up in a variety of different ways:

* _File &#9654; New &#9654; Script..._
* _File &#9654; New &#9654; Text Window_
* _File &#9654; Open..._ on a script file in a supported language
* Drag and drop a script file in a supported language onto the main ImageJ window
* Press the <kbd>[</kbd> shortcut key

We actually make an effort to have the Script Editor be the code editor of choice, but there are still a couple of (quasi-secret/weird) ways for the ImageJ 1.x macro editor window to pop up instead. And we haven't tried extremely hard to quash them because you @abalter points out, there are still a couple of legitimately useful functions that the Script Editor does not have yet.

So in a nutshell, what this issue is really about is just adding the debugging features of the ImageJ 1.x text editor to the ImageJ2 script editor, right?
